modopt/opt/proximity.py

```diff
@@ -434,3 +435,104 @@ def _cost_method(self, *args, **kwargs):
 
         return np.sum([operator.cost(data) for operator, data in
                        zip(self.operators, args[0])])
+
+
+class OrderedWeightedL1Norm(ProximityParent):
+    r"""Ordered Weighted L1 norm proximity operator
+
+    This class defines the OWL proximity operator described in [F2014]
+
+    Parameters
+    ----------
+    weights : np.ndarray
+        Weights values they should be sorted in a non-increasing order
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from modopt.opt.proximity import OrderedWeightedL1Norm
+    >>> A = np.arange(5)*5
+    array([ 0,  5, 10, 15, 20])
+    >>> prox_op = OrderedWeightedL1Norm(np.arange(5))
+    >>> prox_op.weights
+    array([4, 3, 2, 1, 0])
+    >>> prox_op.op(A)
+    array([ 0.,  4.,  8., 12., 16.])
+    >>> prox_op.cost(A, verbose=True)
+     - OWL NORM (X): 150
+    150
+    """
+
+    def __init__(self, weights):
+
+        if any([weights_i < 0 for weights_i in np.squeeze(weights)]):
+            raise ValueError("All the entries of the weights should be"
+                             " positive")
+        self.weights = np.sort(np.squeeze(weights))[::-1]
+        self.op = self._op_method
+        self.cost = self._cost_method
+
+    def _op_method(self, data, extra_factor=1.0):
+        """Operator
+
+        This method returns the input data after the a clustering and a
+        thresholding. Implements (Eq 24) in F2014
+
+        Parameters
+        ----------
+        data : np.ndarray
+            Input data array
+        extra_factor : float
+            Additional multiplication factor
+
+        Returns
+        -------
+        np.ndarray thresholded data
+
+        """
+
+        # Update threshold with extra factor.
+        threshold = self.weights * extra_factor
+
+        # Sorting the absolute value of the input vector
+        data_abs = np.abs(data)
+        idx = np.argsort(np.squeeze(data_abs))[::-1]
+        data_abs = data_abs[idx]
```

Comment:
I think there is something fishy going on there. If I try to do the following:
```python
import numpy as np                                                                                                                     
a = np.array([[2, 7, 8, 9]])                                                                                                           
b = np.abs(a)                                                                                                                          
idx = np.argsort(np.squeeze(b))[::-1]                                                                                                  
b = b[idx]
````

It breaks with `IndexError: index 3 is out of bounds for axis 0 with size 1`. If you squeeze when getting the indexes, then you must squeeze when slicing.

Reply:
Changing it to `flatten`